rule/servlet/src/main/java/io/opentracing/contrib/specialagent/rule/servlet/FilterAgentRule.java

```diff
@@ -38,31 +49,32 @@
         @Override
         public Builder<?> transform(final Builder<?> builder, final TypeDescription typeDescription, final ClassLoader classLoader, final JavaModule module) {
           return builder
-            .visit(Advice.to(ServletInitAdvice.class).on(named("init").and(takesArguments(1)).and(takesArgument(0, named("javax.servlet.ServletConfig")))))
-            .visit(Advice.to(ServletServiceAdvice.class).on(named("service").and(takesArguments(2)).and(takesArgument(0, named("javax.servlet.http.HttpServletRequest")).and(takesArgument(1, named("javax.servlet.http.HttpServletResponse"))))));
+            .visit(Advice.to(ServletInitAdvice.class).on(named("init").and(takesArguments(ServletConfig.class))))
+            .visit(Advice.to(ServletServiceAdvice.class).on(named("service").and(takesArguments(HttpServletRequest.class, HttpServletResponse.class))));
```

Comment:
This refactoring is good, because it is more terse. However, you cannot reference the `ServletConfig`, `HttpServletRequest`, or `HttpServletResponse` classes directly. You must instead reference them by name (i.e. via `named("javax.servlet.ServletConfig")`). This is because it is not this class's responsibility to load classes that belong to the `javax.servlet` package. If you notice, before your changes, there were no imports of classes in the `javax.servlet` package.

Reply:
OK, that's changes things a bit.
I didn't find any method that returns a `ElementMatcher<? super Iterable<? extends TypeDescription>>`.
Do you have a hint?

Reply:
Ah, that makes sense. That's why the code was expressed in the more long form:
`and(takesArguments(1)).and(takesArgument(0, named("javax.servlet.ServletConfig"))))`.

You will have to rewrite the other argument matchers to this pattern as well.